pkg/traceql/engine_metrics.go

```diff
@@ -233,15 +280,22 @@ type TimeSeries struct {
 	Exemplars []Exemplar
 }
 
+type SeriesMapLabel struct {
+	Name  string
+	Value StaticMapKey
+}
+
+type SeriesMapKey [maxGroupBys]SeriesMapLabel
```

Comment:
This is the core change.  SeriesSet changed to use a map key that preserves the static type. This is the same approach used in the `by()` operator itself (GroupingAggregator), but with also the label name.